Source/Core/MemoryEngine/MemoryScanner/MemoryScanner.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+#include <cstring>
+#include <string>
+#include <vector>
+
+#include "../Common/CommonTypes.h"
+#include "../Common/MemoryCommon.h"
+#include "../../Common/Swap.h"
+
+class MemScanner
+{
+public:
+  enum class ScanFiter
+  {
+    exact = 0,
+    increasedBy,
+    decreasedBy,
+    between,
+    biggerThan,
```

Comment:
Same down here, actually (`greaterThan`), but this somewhat depends on whether we want to keep this in the sense of a "library" or just absorb it into Dolphin.